src/components/Personalization/createOnResponseHandler.js

```diff
@@ -9,27 +9,28 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
-import isNonEmptyArray from "../../utils/isNonEmptyArray";
+import isEmptyArray from "../../utils/isEmptyArray";
 
 const DECISIONS_HANDLE = "personalization:decisions";
-
 export default ({ extractDecisions, executeDecisions, showContainers }) => {
   return ({ renderDecisions, response }) => {
     const unprocessedDecisions = response.getPayloadsByType(DECISIONS_HANDLE);
-    if (!isNonEmptyArray(unprocessedDecisions)) {
+
+    if (!renderDecisions) {
+      return { decisions: unprocessedDecisions };
+    }
+
+    if (isEmptyArray(unprocessedDecisions)) {
```

Comment:
unprocessedDecisions will always be an array, so you can just do

if (unprocessedDecisions.length === 0) {

Reply:
thank you